samples/Sentry.Samples.OpenTelemetry.AspNetCore/Program.cs

```diff
@@ -0,0 +1,56 @@
+using System.ComponentModel.DataAnnotations;
+using System.Diagnostics;
+using OpenTelemetry.Resources;
+using OpenTelemetry.Trace;
+using Sentry;
+using Sentry.OpenTelemetry;
+using Sentry.Samples.OpenTelemetry.AspNetCore;
+
+var builder = WebApplication.CreateBuilder(args);
+
+// OpenTelemetry Configuration
+// See https://opentelemetry.io/docs/instrumentation/net/getting-started/
+builder.Services.AddOpenTelemetry()
+    .WithTracing(tracerProviderBuilder =>
+        tracerProviderBuilder
+            .AddSource(DiagnosticsConfig.ActivitySource.Name)
+            .ConfigureResource(resource => resource.AddService(DiagnosticsConfig.ServiceName))
+            .AddAspNetCoreInstrumentation()
+            .AddHttpClientInstrumentation()
+            .AddSentry()
+        );
+
+builder.WebHost.UseSentry(options =>
+{
+    options.TracesSampleRate = 1.0;
+    options.UseOpenTelemetry();
```

Comment:
This looks good as init code. But do we override someone's propagator if they set something?

Reply:
We would yes... that happens in the `AddSentry()` call. See the [comments on that extension](https://github.com/getsentry/sentry-dotnet/blob/46490d770d5d9b4539f51365b1b10738561cffa5/src/Sentry.OpenTelemetry/TracerProviderBuilderExtensions.cs#L23):
> The default TextMapPropagator to be used by OpenTelemetry.
>
> If this parameter is not supplied, the <see cref="SentryPropagator"/> will be used, which propagates the baggage header as well as Sentry trace headers.
>
> The <see cref="SentryPropagator"/> is required for Sentry's OpenTelemetry integration to work but you could wrap this in a `CompositeTextMapPropagator` if you needed other propagators as well.

Under the hood the OpenTelemetry SDK just [stores this as a static property](https://github.com/open-telemetry/opentelemetry-dotnet/blob/104cf32d97b2aba7319635fc23d84d5f7663fe9b/src/OpenTelemetry.Api/Context/Propagation/Propagators.cs#L32), so I think the composite propagator is the only way around it. We could try to do something fancy like, if this was set to something other than `Noop`, we read that out and stuff it in a composite property along with our `SentryPropagator` and then write it back... essentially emulating a `List<TextMapPropagator>`.